polars/polars-arrow/src/kernels/rolling/no_nulls.rs

```diff
@@ -73,6 +89,75 @@ where
     ))
 }
 
+fn rolling_apply_quantile<T, Fo, Fa>(
+    values: &[T],
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    det_offsets_fn: Fo,
+    aggregator: Fa,
+) -> ArrayRef
+where
+    Fo: Fn(Idx, WindowSize, Len) -> (Start, End),
+    Fa: Fn(&[T], f64, QuantileInterpolOptions) -> T,
+    T: Debug + NativeType,
+{
+    let len = values.len();
+    let out = (0..len)
+        .map(|idx| {
+            let (start, end) = det_offsets_fn(idx, window_size, len);
+            let vals = unsafe { values.get_unchecked(start..end) };
+            aggregator(vals, quantile, interpolation)
+        })
+        .collect_trusted::<Vec<T>>();
+
+    let validity = create_validity(min_periods, len as usize, window_size, det_offsets_fn);
+    Arc::new(PrimitiveArray::from_data(
+        T::PRIMITIVE.into(),
+        out.into(),
+        validity.map(|b| b.into()),
+    ))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn rolling_apply_convolve_quantile<Fo, Fa>(
+    values: &[f64],
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    det_offsets_fn: Fo,
+    aggregator: Fa,
+    weights: &[f64],
+) -> ArrayRef
+where
+    Fo: Fn(Idx, WindowSize, Len) -> (Start, End),
+    Fa: Fn(&[f64], f64, QuantileInterpolOptions) -> f64,
+{
+    assert_eq!(weights.len(), window_size);
+    let mut buf = vec![0.0; window_size];
+    let len = values.len();
+    let out = (0..len)
+        .map(|idx| {
+            let (start, end) = det_offsets_fn(idx, window_size, len);
+            let vals = unsafe { values.get_unchecked(start..end) };
+            buf.iter_mut()
+                .zip(vals.iter().zip(weights))
+                .for_each(|(b, (v, w))| *b = *v * *w);
+
+            aggregator(&buf, quantile, interpolation)
+        })
+        .collect_trusted::<Vec<f64>>();
+
+    let validity = create_validity(min_periods, len as usize, window_size, det_offsets_fn);
+    Arc::new(PrimitiveArray::from_data(
```

Comment:
This also should work on both `f32` and `f64` data. I think you can use the same solution as you did on polars/polars-arrow/src/kernels/rolling/nulls.rs  line 63